src/main/java/slimeknights/tconstruct/smeltery/block/CastingTankBlock.java

```diff
@@ -0,0 +1,151 @@
+  protected static final DirectionProperty FACING = BlockStateProperties.HORIZONTAL_FACING;
+  private final PushReaction pushReaction;
+  public CastingTankBlock(Properties properties, PushReaction pushReaction) {
+    super(properties);
+    this.pushReaction = pushReaction;
+    registerDefaultState(stateDefinition.any().setValue(FACING, Direction.NORTH));
+  }
+
+  public CastingTankBlock(Properties properties) {
+    this(properties, PushReaction.BLOCK);
+  }
+
+  @Override
+  public BlockState getStateForPlacement(BlockPlaceContext context) {
+    return defaultBlockState().setValue(FACING, context.getHorizontalDirection().getOpposite());
+  }
+
+  @Deprecated
+  @Override
+  public BlockState rotate(BlockState state, Rotation rot) {
+    return state.setValue(FACING, rot.rotate(state.getValue(FACING)));
+  }
+
+  @Deprecated
+  @Override
+  public BlockState mirror(BlockState state, Mirror mirrorIn) {
+    return state.rotate(mirrorIn.getRotation(state.getValue(FACING)));
+  }
+
+  @Override
+  protected void createBlockStateDefinition(StateDefinition.Builder<Block, BlockState> builder) {
+    builder.add(FACING);
+  }
+
+  @Override
+  protected boolean openGui(Player player, Level world, BlockPos pos) {
+    return false;
+  }
+
+  @Override
+  public PushReaction getPistonPushReaction(BlockState pState) {
+    return pushReaction;
+  }
+
+  @Deprecated
+  @Override
+  public float getShadeBrightness(BlockState state, BlockGetter worldIn, BlockPos pos) {
+    return 1.0F;
+  }
+
+  @Override
+  @Nullable
+  public BlockEntity newBlockEntity(BlockPos pPos, BlockState pState) {
+    return new CastingTankBlockEntity(pPos, pState, this);
+  }
+
+  @Deprecated
+  @Override
+  public InteractionResult use(BlockState state, Level world, BlockPos pos, Player player, InteractionHand hand, BlockHitResult hit) {
+    if (player.isSuppressingBounce()) {
+      return InteractionResult.PASS;
+    }
+
+    BlockEntity te = world.getBlockEntity(pos);
+    if (te instanceof CastingTankBlockEntity) {
+      ((CastingTankBlockEntity) te).interact(player, hand, hit);
+      return InteractionResult.SUCCESS;
+    }
+
+    return super.use(state, world, pos, player, hand, hit);
+  }
+
+  @Override
+  public int getLightEmission(BlockState state, BlockGetter world, BlockPos pos) {
+    BlockEntity te = world.getBlockEntity(pos);
+    if (te instanceof ITankBlockEntity) {
+      FluidStack fluid = ((ITankBlockEntity) te).getTank().getFluid();
+      return fluid.getFluid().getFluidType().getLightLevel(fluid);
+    }
+    return super.getLightEmission(state, world, pos);
+  }
+
+  @Override
+  public void setPlacedBy(Level worldIn, BlockPos pos, BlockState state, @Nullable LivingEntity placer, ItemStack stack) {
+    CompoundTag nbt = stack.getTag();
+    if (nbt != null) {
+      BlockEntityHelper.get(CastingTankBlockEntity.class, worldIn, pos).ifPresent(te -> te.updateTank(nbt.getCompound(NBTTags.TANK)));
```

Comment:
I'm inclined to just ditch `BlockEntityHelper` from our code, just do the standard if statement like the other methods.